contracts/src/v0.8/dev/VRFV2Wrapper.sol

```diff
@@ -0,0 +1,260 @@
+   *        function.
+   * @param _wrapperPremiumPercentage is the premium ratio in percentage for wrapper requests.
+   * @param _keyHash to use for requesting randomness.
+   */
+  function setConfig(
+    uint32 _wrapperGasOverhead,
+    uint8 _wrapperPremiumPercentage,
+    bytes32 _keyHash,
+    uint8 _maxNumWords
+  ) external onlyOwner {
+    s_wrapperGasOverhead = _wrapperGasOverhead;
+    s_wrapperPremiumPercentage = _wrapperPremiumPercentage;
+    s_keyHash = _keyHash;
+    s_maxNumWords = _maxNumWords;
+    s_configured = true;
+
+    // Get other configuration from coordinator
+    (, , s_stalenessSeconds, s_gasAfterPaymentCalculation) = COORDINATOR.getConfig();
+    s_fallbackWeiPerUnitLink = COORDINATOR.getFallbackWeiPerUnitLink();
+  }
+
+  /**
+   *
+   */
+  function getConfig()
+    external
+    view
+    returns (
+      int256 fallbackWeiPerUnitLink,
+      uint32 stalenessSeconds,
+      uint32 fulfillmentFlatFeeLinkPPM,
+      uint32 gasAfterPaymentCalculation,
+      uint32 wrapperGasOverhead,
+      uint8 wrapperPremiumPercentage,
+      bytes32 keyHash,
+      uint8 maxNumWords
+    )
+  {
+    return (
+      s_fallbackWeiPerUnitLink,
+      s_stalenessSeconds,
+      s_fulfillmentFlatFeeLinkPPM,
+      s_gasAfterPaymentCalculation,
+      s_wrapperPremiumPercentage,
+      s_wrapperPremiumPercentage,
+      s_keyHash,
+      s_maxNumWords
+    );
+  }
+
+  function calculatePrice(uint32 callbackGasLimit) public view override onlyConfiguredNotDisabled returns (uint256) {
+    return calculatePriceInternal(callbackGasLimit, tx.gasprice);
+  }
+
+  function estimatePrice(uint32 _callbackGasLimit, uint256 _requestGasPrice)
+    external
+    view
+    onlyConfiguredNotDisabled
+    returns (uint256)
+  {
+    return calculatePriceInternal(_callbackGasLimit, _requestGasPrice);
+  }
+
+  function calculatePriceInternal(uint32 _callbackGasLimit, uint256 _requestGasPrice) internal view returns (uint256) {
+    int256 weiPerUnitLink = getFeedData();
+    require(weiPerUnitLink >= 0, "Invalid LINK wei price");
+
+    uint256 feeNoWrapperPremium = (1e18 * _requestGasPrice * (_callbackGasLimit + s_wrapperGasOverhead)) /
+      uint256(weiPerUnitLink) +
+      (1e12 * uint256(s_fulfillmentFlatFeeLinkPPM));
+
+    uint256 fee = (feeNoWrapperPremium * (s_wrapperPremiumPercentage + 100)) / 100;
+
+    return fee;
+  }
+
+  function getFeedData() private view returns (int256) {
+    bool staleFallback = s_stalenessSeconds > 0;
+    uint256 timestamp;
+    int256 weiPerUnitLink;
+    (, weiPerUnitLink, , timestamp, ) = LINK_ETH_FEED.latestRoundData();
+    // solhint-disable-next-line not-rely-on-time
+    if (staleFallback && s_stalenessSeconds < block.timestamp - timestamp) {
+      weiPerUnitLink = s_fallbackWeiPerUnitLink;
+    }
+    return weiPerUnitLink;
+  }
+
+  function onTokenTransfer(
+    address _sender,
+    uint256 _amount,
+    bytes calldata _data
+  ) external onlyConfiguredNotDisabled {
```

Comment:
Very cool to leverage transferAndCall as a direct request mechanism.

Reply:
Yeah we do this in a lot of spots, I like the pattern. VRF V1 does this too